Sources/BraintreePayPalMessaging/BTPayPalMessagingClient.swift

```diff
@@ -0,0 +1,150 @@
+public class BTPayPalMessagingClient: UIView {
+
+    // MARK: - Properties
+
+    public weak var delegate: BTPayPalMessagingDelegate?
+
+    var apiClient: BTAPIClient
+
+    // MARK: - Initializers
+
+    ///  Initializes a PayPal Messaging client.
+    /// - Parameter apiClient: The Braintree API client
+    public init(apiClient: BTAPIClient) {
+        self.apiClient = apiClient
+
+        super.init(frame: .zero)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Public Method
+
+    /// Creates a `BTPayPalMessagingClient` to be displayed to promote offers such as Pay Later and PayPal Credit to customers.
+    /// - Parameter request: an optional `BTPayPalMessagingRequest`
+    public func createView(_ request: BTPayPalMessagingRequest = BTPayPalMessagingRequest()) {
+        apiClient.fetchOrReturnRemoteConfiguration { configuration, error in
+            if let error {
+                self.delegate?.onError(self, error: error)
+                return
+            }
+
+            guard let configuration else {
+                self.delegate?.onError(self, error: BTPayPalMessagingError.fetchConfigurationFailed)
+                return
+            }
+
+            guard let clientID = configuration.json?["paypal"]["clientId"].asString() else {
+                self.delegate?.onError(self, error: BTPayPalMessagingError.payPalClientIDNotFound)
+                return
+            }
+
+            let messageData = PayPalMessageData(
+                clientID: clientID,
+                environment: configuration.environment == "production" ? .live : .sandbox,
+                amount: request.amount,
+                placement: request.placement?.placementRawValue,
+                offerType: request.offerType?.offerTypeRawValue
+            )
+
+            messageData.buyerCountry = request.buyerCountry
+
+            let messageConfig = PayPalMessageConfig(
+                data: messageData,
+                style: PayPalMessageStyle(
+                    logoType: request.logoType.logoTypeRawValue,
+                    color: request.color.messageColorRawValue,
+                    textAlignment: request.textAlignment.textAlignmentRawValue
+                )
+            )
+
+            let messageView = PayPalMessageView(config: messageConfig, stateDelegate: self, eventDelegate: self)
+            messageView.translatesAutoresizingMaskIntoConstraints = false
+            self.addSubview(messageView)
+
+            NSLayoutConstraint.activate([
+                messageView.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+                messageView.trailingAnchor.constraint(equalTo: self.trailingAnchor),
+                messageView.topAnchor.constraint(equalTo: self.topAnchor),
+                messageView.bottomAnchor.constraint(equalTo: self.bottomAnchor)
+            ])
```

Comment:
Here we're doing UI work off the main thread (in the callback from `fetchOrReturnRemoteConfiguration`) - I guess since this view isn't yet set onscreen (which the merchant would have to do), this doesn't cause an issue? Just wanted to call this out.

Reply:
Another thing - these operations happen async since we're blocked by the config fetch. With the current design, a merchant needs to wait for this to `createView()` call to complete the necessary operations before displaying it onscreen.

I think we need to provide a hook into the completion of this method, or else a merchant could proceed before those constraints are set and nothing would load.

Reply:
Another thing, can we make it more clear in our docstrings where a merchant has to hook into the result of this `createView()` call (`BTPayPalMessagingDelegate`).

Reply:
> I guess since this view isn't yet set onscreen (which the merchant would have to do), this doesn't cause an issue? Just wanted to call this out.

Correct, this will adjust view layout based on provided constraints once the view created and does not need to be called on the main thread.

Reply:
> Another thing, can we make it more clear in our docstrings where a merchant has to hook into the result of this createView() call (BTPayPalMessagingDelegate).

Good call, updated: https://github.com/braintree/braintree_ios/pull/1153/commits/8463a60b5ed3ff8dd651c4788ac5455e7cbf4c1d